src/dbnode/client/fetch_state.go

```diff
@@ -231,7 +235,9 @@ func (f *fetchState) asEncodingSeriesIterators(
 		return nil, FetchResponseMetadata{}, err
 	}
 
-	limit := f.fetchTaggedOp.requestSeriesLimit(maxInt)
```

Comment:
this was the bug :( it was using the series limit on the req when accumulating results. this was ok previously when the req.SeriesLimit == query.SeriesLimit